lib/bindings/MQTTBinding.js

```diff
@@ -0,0 +1,335 @@
+        }
+    }
+
+    if (attribute === constants.TIMESTAMP_ATTRIBUTE) {
+        return constants.TIMESTAMP_TYPE;
+    } else {
+        return constants.DEFAULT_ATTRIBUTE_TYPE;
+    }
+}
+
+/**
+ * Get the API Key for the selected service if there is any, or the default API Key if a specific one does not exist.
+ *
+ * @param {String} service          Name of the service whose API Key we are retrieving.
+ * @param {String} subservice       Name of the subservice whose API Key we are retrieving.
+ */
+function getEffectiveApiKey(service, subservice, callback) {
+    logger.debug(context, 'Getting effective API Key');
+
+    iotAgentLib.findConfiguration(service, subservice, function(error, group) {
+        if (group) {
+            logger.debug('Using found group: %j', group);
+            callback(null, group.apikey);
+        } else if (config.mqtt.defaultKey) {
+            logger.debug('Using default API Key: %s', config.mqtt.defaultKey);
+            callback(null, config.mqtt.defaultKey);
+        } else {
+            logger.error(context, 'Could not find any API Key information for device.');
+            callback(new errors.GroupNotFound(service, subservice));
+        }
+    });
+}
+
+/**
+ * Generate the list of topics related to the device, based on the device attribute definitions.
+ *
+ * @param {Object} device           Device object containing all the information about the provisioned device.
+ * @param {String} apikey           API Key the device is subscribed to.
+ */
+function generateDeviceTopics(device, apikey, callback) {
+    var topics = [];
+
+    logger.debug(context, 'Generating device topics');
+    topics.push('/' + apikey + '/' + device.id + '/' + constants.MEASURES_SUFIX + '/+');
+    topics.push('/' + apikey + '/' + device.id + '/' + constants.MEASURES_SUFIX);
+    topics.push('/' + apikey + '/' + device.id + '/' + constants.CONFIGURATION_SUFIX +
+        '/' + constants.CONFIGURATION_COMMAND_SUFIX);
+
+    callback(null, topics);
+}
+
+function deviceProvisioningHandler(device, callback) {
+    function subscribeToTopics(topics, callback) {
+        logger.debug('Subscribing to topics');
+
+        mqttClient.subscribe(topics, null, function(error) {
+            if (error) {
+                logger.error('Error subscribing to device topics: %s', error);
+                callback(error);
+            } else {
+                logger.debug('Successfully subscribed to the following topics:\n%j\n', topics);
+                callback(null, device);
+            }
+        });
+    }
+
+    if (mqttClient) {
+        async.waterfall([
+            apply(getEffectiveApiKey, device.service, device.subservice),
+            apply(generateDeviceTopics, device),
+            subscribeToTopics
+        ], callback);
+    } else {
+        callback();
+    }
+}
+
+function configurationHandler(configuration, callback) {
+
+}
+
+function notificationHandler(device, updates, callback) {
```

Comment:
It will be filled in future PRs, but I'll remove it from the branch temporarily, as it is useless right now.